plugin.py

```diff
@@ -129,7 +185,154 @@ def run(self, edit: sublime.Edit) -> None:
         if session is None:
             return
 
-        session.send_request(Request("rust-analyzer/reloadWorkspace"))
+        session.send_request(Request("rust-analyzer/reloadWorkspace"), self.on_result)
+
+    def on_result(self, arg: Any):
+        pass
+
+
+class RustAnalyzerMemoryUsage(LspTextCommand):
+    session_name = "rust-analyzer"
+
+    def run(self, edit: sublime.Edit) -> None:
+        session = self.session_by_name(self.session_name)
+        if session is None:
+            return
+
+        session.send_request(Request("rust-analyzer/memoryUsage"), self.on_result)
+
+    def on_result(self, payload: str) -> None:
+        window = self.view.window()
+        if window is None:
+            return
+        sheets = window.selected_sheets()
+        view = window.new_file(flags=sublime.TRANSIENT)
+        view.set_scratch(True)
+        view.set_name("--- RustAnalyzer Memory Usage ---")
+        view.run_command("append", {"characters": "Per-query memory usage:"})
+        for line in payload.splitlines():
+            view.run_command("append", {"characters": '\n{}'.format(line)})
+        view.run_command("append", {"characters": "\n(note: database has been cleared)"})
+        view.set_read_only(True)
+        sheet = view.sheet()
+        if sheet is not None:
+            sheets.append(sheet)
+            window.select_sheets(sheets)
```

Comment:
Same comment as in the other review... I'm wondering if the sheet handling here is correct because I don't see any difference from just creating a new file.

Reply:
This creates a side-by-side view similar to how vscode does it. very elegant!

Reply:
It doesn't seem to be entirely correct though since it collects all sheets from all groups and then selects them here.

(I guess this could be fixed later).